src/AsmResolver.PE.Win32Resources/Icon/IconGroupDirectory.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace AsmResolver.PE.Win32Resources.Icon
+{
+    /// <summary>
+    /// Represents a single win32 icon group directory.
+    /// </summary>
+    public class IconGroupDirectory : SegmentBase
+    {
+        /// <summary>
+        /// Used to keep track of all icon entries associated with this icon group.
+        /// </summary>
+        private readonly IDictionary<ushort, Tuple<IconGroupDirectoryEntry, IconEntry>> _entries = new Dictionary<ushort, Tuple<IconGroupDirectoryEntry, IconEntry>>();
```

Comment:
Use `(IconGroupDirectoryEntry, IconEntry)`, instead of Tuple to reduce heap allocations.